tsdb/engine/tsm1/engine.go

```diff
@@ -228,7 +258,8 @@ func (e *DevEngine) Begin(writable bool) (tsdb.Tx, error) {
 
 func (e *DevEngine) WriteTo(w io.Writer) (n int64, err error) { panic("not implemented") }
 
-func (e *DevEngine) writeSnapshot() error {
+// WriteSnapshot will snapshot the cache and write a new TSM file with its contents, releasing the snapshot when done.
+func (e *DevEngine) WriteSnapshot(async bool) error {
```

Comment:
Yeah, I'm not keen on this.  I would like see some data or benchmarks that can point to a problem that this is solving.  The first goroutine introduced a race condition after it was merged yesterday and I'm not sure what unintended issues this could introduce.